src/main/java/com/lgtm/easymoney/services/impl/FriendServiceImpl.java

```diff
@@ -38,45 +38,43 @@ public FriendServiceImpl(UserService userService, FriendshipRepository friendshi
   }
 
   @Override
-  public Friendship getFriendshipRecord(User u1, User u2) {
-    return friendshipRepository.findByUser1AndUser2(u1, u2);
-  }
-
-  @Override
-  public void addFriend(FriendshipReq friendshipReq) {
+  public void checkUserType(FriendshipReq friendshipReq) {
     User requester = userService.getUserById(friendshipReq.getUid1());
     if (requester.getType() != UserType.PERSONAL) {
       throw new InapplicableOperationException(
-          "User", friendshipReq.getUid1(), "uid1", "addFriend");
+              "User", friendshipReq.getUid1(), "uid1", "addFriend");
     }
     User acceptor = userService.getUserById(friendshipReq.getUid2());
     if (acceptor.getType() != UserType.PERSONAL) {
       throw new InapplicableOperationException(
-          "User", friendshipReq.getUid2(), "uid2", "addFriend");
+              "User", friendshipReq.getUid2(), "uid2", "addFriend");
     }
```

Comment:
I think this function can just return the two user objects directly, so you don't need to run getUserById below again as the DB queries involved can be expensive. Or alternatively, just write a function that gets and checks one user, so that `getFriends()` and `getFriendsPending()` can also use it